### ISSUE TITLE
Question about producer and deliveryChan

### ISSUE THREAD

**Issue by OsmanAktemur** [confluentinc/confluent-kafka-go#1262]
Description
===========

In the examples of producers there is usage of deliveryChan and wait for the channel. So i think this makes async operation to sync and wait for message sent to broker successfuly.

My question is it possible to wait for deliveryChan never return any data and blocks forever? Maybe because some kind of network failure etc?  Should we implement any timeout  for **e := <-deliveryChan**   ? 
 
	err = p.Produce(&kafka.Message{
		TopicPartition: kafka.TopicPartition{Topic: &topic, Partition: kafka.PartitionAny},
		Value:          payload,
		Headers:        []kafka.Header{{Key: "myTestHeader", Value: []byte("header values are binary")}},
	}, deliveryChan)
	if err != nil {
		fmt.Printf("Produce failed: %v\n", err)
		os.Exit(1)
	}

	e := <-deliveryChan
	m := e.(*kafka.Message)

	if m.TopicPartition.Error != nil {
		fmt.Printf("Delivery failed: %v\n", m.TopicPartition.Error)
	} else {
		fmt.Printf("Delivered message to topic %s [%d] at offset %v\n",
			*m.TopicPartition.Topic, m.TopicPartition.Partition, m.TopicPartition.Offset)
	}

https://github.com/confluentinc/confluent-kafka-go/blob/master/examples/json_producer_example/json_producer_example.go

Checklist
=========
Please provide the following information:

 - [ ] confluent-kafka-go and librdkafka version (`LibraryVersion()`):
 - [ ] Apache Kafka broker version:
 - [ ] Client configuration: `ConfigMap{...}`
 - [ ] Operating system:
 - [ ] Provide client logs (with `"debug": ".."` as necessary)
 - [ ] Provide broker log excerpts
 - [ ] Critical issue

**Comment by milindl** [comment]
No, the delivery channel should not block for more than the value of the property set by the key "delivery.timeout.ms". Note that this can be set to be "infinite" (if it's set to 0), but barring that value, the message will time out after this time and deliver and error report on the delivery channel. The default is 30 seconds.

Also, note that you should be polling the producer.Events() channel in a separate go-routine regardless of whether you are making the producer "sync".

**Comment by OsmanAktemur** [comment]
Thank you @milindl

**Comment by milindl** [comment]
You're welcome. I'm closing this, don't hesitate to open issues for more questions.
